pkg/controllers/node/ipam.go

```diff
@@ -50,17 +53,20 @@ func (c *NodeController) syncIPAMCleanup() error {
 	c.rl.Forget(RateLimitCalicoList)
 
 	// Build a list of all the nodes in the cluster based on IPAM allocations across all
-	// blocks, plus affinities.
+	// blocks, plus affinities. Entries in the 'nodes' map will be indexed by the name of the
+	// node in the Kubernetes API.
 	nodes := map[string][]model.AllocationAttribute{}
 	for _, kvp := range blocks.KVPairs {
 		b := kvp.Value.(*model.AllocationBlock)
 
 		// Include affinity if it exists. We want to track nodes even
 		// if there are no IPs actually assigned to that node.
 		if b.Affinity != nil {
-			n := strings.TrimPrefix(*b.Affinity, "host:")
-			if _, ok := nodes[n]; !ok {
-				nodes[n] = []model.AllocationAttribute{}
+			n := c.kubernetesNodeForCalico(strings.TrimPrefix(*b.Affinity, "host:"))
```

Comment:
This method seems to return either k8s node name or a calico node name depending on what's available.

Could we make a little clearer by:
- Keeping this processing unchanged, perhaps just changing nodes -> calicoNodes to make it clearer
- When we loop through allocations we replace the nodeExists() with something like getKubernetesNodeFromCalico() and then we keep all of the k8s node stuff just within that loop.  It also means the call to cleanupNode can be made with the correct node name (which I think with this code it isn't).

Reply:
^ note the second bullet.  I think the call to cleanupNode is called with the wrong node name.

Reply:
Also perhaps a good opportunity to tighten up naming when we use the name node to make it clear whether calico or kubernetes.